apps/src/tutorialExplorer/README.md

```diff
@@ -0,0 +1,119 @@
+
+## Variables
+
+Three [DCDO](https://github.com/code-dot-org/code-dot-org/blob/929fb8fffe8c70817630627df5766bf2c706d9fc/lib/dynamic_config/dcdo.rb) [variables](https://github.com/code-dot-org/code-dot-org/blob/929fb8fffe8c70817630627df5766bf2c706d9fc/pegasus/sites.v3/code.org/public/learn/index.haml#L145-L147) can be used to vary the behaviour of the **Tutorial Explorer** without requiring a deploy to the site:
+
+- `learn_show_sort_dropdown`: Whether the "Sort by" dropdown should be shown at all.  Defaults to `true`.
+- `learn_hide_tutorials`: An array of tutorial "short codes" that should be hidden.  Useful if a tutorial suddenly becomes unavailable, say due to a server issue.  Defaults to `[]`.
+- `defaultSortByPopularity`: Whether the "Sort by" option defaults to sorting by "Most popular", rather than "Recommended".  The default here actually varies by site and current `hoc_mode`, with the logic captured [here](https://github.com/code-dot-org/code-dot-org/blob/929fb8fffe8c70817630627df5766bf2c706d9fc/pegasus/src/database.rb#L86-L88).
+
+## Analytics
+
+When a tutorial is started, we open it in a new browser tab.  Since the **Tutorial Explorer** tab remains in existence, we simply generate a couple Google Analytics events at the same time, as seen [here](https://github.com/code-dot-org/code-dot-org/blob/929fb8fffe8c70817630627df5766bf2c706d9fc/apps/src/tutorialExplorer/tutorialDetail.jsx#L47-L48).
+
+## Mobile detection
+
+There is some rudimentary [detection for mobile devices](https://github.com/code-dot-org/code-dot-org/blob/929fb8fffe8c70817630627df5766bf2c706d9fc/apps/src/tutorialExplorer/util.jsx#L108-L132).  If a mobile device is successfully detected, the Android & iOS options are initially checked.
+
+## URL parameters
+
+It is possible to pass a custom URL to `/learn` to pre-select the filters shown.
+
+Here's a simple example:
+```
+https://hourofcode.com/learn?platform=no-computers
+```
+
+This says that for filter `platform`, entry `no-computers` should be selected.
+
+It's also possible to specify multiple filter entries, even for the same filter, for example:
+```
+https://hourofcode.com/learn?grade=pre&grade=2-5&platform=no-computers
+```
+
+Here is a full list of the filter names & entries that can be specified:
+```
+grade: pre, 2-5, 6-8, 9+
+teacher_experience: beginner, comfortable
+student_experience: beginner, comfortable
+platform: computers, android, ios, no-internet, no-computers
+subject: science, math, history, la, art, cs-only
+activity_type: online-tutorial, lesson-plan
+length: 1hour, 1hour-follow, few-hours
+programming_language: blocks, typing, other
+```
+
+**Important note:** In order to ensure satisfactory web performance, list these parameters in the order of the above list.  (This will allow each variant to be cached as one.)
+
+For example, do not specify: https://hourofcode.com/learn?subject=science&platform=computers
+
+Instead, specify: https://hourofcode.com/learn?platform=computers&subject=science
```

Comment:
Yay! Thanks for the clarifying examples throughout.

Reply:
Fortunately I had written this up in an email long ago, so it was easy to transfer!